instrumentation/kamon-redis/src/main/scala/kamon/instrumentation/jedis/JedisInstrumentation.scala

```diff
@@ -8,27 +8,29 @@ import redis.clients.jedis.commands.ProtocolCommand
 
 class JedisInstrumentation extends InstrumentationBuilder {
   onType("redis.clients.jedis.Protocol")
-    .advise(method("sendCommand").and(withArgument(1, classOf[ProtocolCommand])), classOf[SendCommandAdvice])
+    .advise(method("sendCommand"), classOf[SendCommandAdvice])
 }
 
 class SendCommandAdvice
-
 object SendCommandAdvice {
   @Advice.OnMethodEnter()
-  def enter(@Advice.Argument(1) command: ProtocolCommand) = {
-    val spanName = s"redis.command.${command}"
-    val span = Kamon.clientSpanBuilder(spanName, "redis.client.jedis")
-      .start()
-
-    span
+  def enter(@Advice.Argument(1) command: Any) = {
+    command match {
+      case command: ProtocolCommand =>
+        val spanName = s"redis.command.${command}"
+        Some(Kamon.clientSpanBuilder(spanName, "redis.client.jedis")
+          .start())
+      case _ => None
```

Comment:
Tiny thing: you could use `Span.Empty` to avoid creating options altogether, specially because you know that 100% of the time the argument will be `ProtocolCommand`. If you return `Span` instead of `Option[Span]` the exit would be exactly the same as it was before, and it will do nothing if it ever gets `Span.Empty`